src/Analyzers/Core/CodeFixes/ImplementInterface/AbstractImplementInterfaceCodeFixProvider.cs

```diff
@@ -39,173 +27,11 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         if (!token.Span.IntersectsWith(span))
             return;
 
-        var options = await document.GetImplementTypeOptionsAsync(cancellationToken).ConfigureAwait(false);
-
-        foreach (var type in token.Parent.GetAncestorsOrThis<TTypeSyntax>())
-        {
-            if (this.IsTypeInInterfaceBaseList(type))
-            {
-                var service = document.GetRequiredLanguageService<IImplementInterfaceService>();
-
-                var info = await service.AnalyzeAsync(
-                    document, type, cancellationToken).ConfigureAwait(false);
-                if (info is not null)
-                {
-                    using var _ = ArrayBuilder<CodeAction>.GetInstance(out var codeActions);
-                    await foreach (var implementOptions in GetImplementOptionsAsync(document, info, cancellationToken))
-                    {
-                        var title = GetTitle(implementOptions);
-                        var equivalenceKey = GetEquivalenceKey(info, implementOptions);
-                        codeActions.Add(CodeAction.Create(
-                            title,
-                            cancellationToken => service.ImplementInterfaceAsync(
-                                document, info, options, implementOptions, cancellationToken),
-                            equivalenceKey));
-                    }
-
-                    context.RegisterFixes(codeActions, context.Diagnostics);
-                }
-
-                break;
-            }
-        }
-    }
-
-    private static string GetTitle(ImplementInterfaceConfiguration options)
```

Comment:
all teh below was moved from the codefixprovider to the service.  so it can be used by teh refactoring as well.